File: src/main/java/com/github/wgzhao/dbquery/param/RestResponse.java
package com.github.wgzhao.dbquery.param;

import org.apache.commons.lang3.ArrayUtils;

import java.io.Serializable;

public class RestResponse<T>
        implements Serializable
{
    private static final long serialVersionUID = 691617930352197222L;

    private static final Integer[] successStatus = new Integer[] {200};

    private Integer status;

    private String message;

    private Long total;

    private T data;

    public RestResponse() {}

    public RestResponse(Integer status, String message)
    {
        this.status = status;
        this.message = message;
    }

    public RestResponse(Integer status, String message, Long total, T data)
    {
        this.status = status;
        this.message = message;
        this.total = total;
        this.data = data;
    }

    public Integer getStatus()
    {
        return this.status;
    }

    public void setStatus(Integer status)
    {
        this.status = status;
    }

    public Boolean isSuccess()
    {
        return (ArrayUtils.indexOf(successStatus, this.status) >= 0);
    }

    public String getMessage()
    {
        return this.message;
    }

    public void setMessage(String message)
    {
        this.message = message;
    }

    public T getData()
    {
        return this.data;
    }

    public void setData(T data)
    {
        this.data = data;
    }

    public Long getTotal()
    {
        return this.total;
    }

    public void setTotal(Long total)
    {
        this.total = total;
    }
}

File: src/main/kotlin/com/github/wgzhao/dbquery/param/RestResponse.kt
package com.github.wgzhao.dbquery.param

import org.apache.commons.lang3.ArrayUtils
import java.io.Serializable

class RestResponse<T>
    : Serializable {
    var status: Int? = null

    var message: String? = null

    var total: Long? = null

    var data: T? = null

    constructor()

    constructor(status: Int?, message: String?) {
        this.status = status
        this.message = message
    }

    constructor(status: Int?, message: String?, total: Long?, data: T?) {
        this.status = status
        this.message = message
        this.total = total
        this.data = data
    }

    val isSuccess: Boolean
        get() = (ArrayUtils.indexOf(successStatus, this.status) >= 0)

    companion object {
        private const val serialVersionUID = 691617930352197222L

        private val successStatus: Array<Int?> = arrayOf<Int>(200)
    }
}
